src/SFA.DAS.LearnerDataMismatches.Web/Pages/Learner.cshtml.cs

```diff
@@ -74,7 +80,13 @@ private async Task BuildNewCollections(long learnerUln)
                 .Where(x => x.LearnerUln == learnerUln)
                 .ToListAsync();
 
-            var report = new LearnerReport(apprenticeships, earnings);
+            var locks = await context.DataLockgEvent
+                .Include(x => x.NonPayablePeriods)
+                .ThenInclude(x => x.DataLockEventNonPayablePeriodFailures)
+                .Where(x => x.LearnerUln == learnerUln)
+                .ToListAsync();
+
+            var report = new LearnerReport(apprenticeships, earnings, locks);
 
             NewCollectionPeriods = report.CollectionPeriods;
```

Comment:
Ha, a nice higlight to my other "keep it in the domain" comments! 👍 

I was probably thinking that display ordering belongs in the ViewModel rather than the domain.  However I'm happy to move this.